drivers/wireless/bluetooth/bt_rpmsghci_server.c

```diff
@@ -0,0 +1,538 @@
+{
+  FAR struct rpmsghci_server_s *server = NULL;
+  FAR struct bt_driver_s       *btdev  = NULL;
+  FAR struct rpmsghci_data_s   *msg    = data;
+  int                           ret    = OK;
+
+  DEBUGASSERT(priv);
+  server = priv;
+  DEBUGASSERT(server->btdev);
+  btdev = server->btdev;
+
+  if (btdev->close)
+    {
+      btdev->close(btdev);
+    }
+
+  msg->header.result = ret;
+
+  return rpmsg_send(ept, msg, sizeof(struct rpmsghci_header_s));
+}
+
+/****************************************************************************
+ * Name: rpmsghci_send_handler
+ *
+ * Description:
+ *   Rpmsg-HCI send handler.
+ *
+ * Parameters:
+ *   ept  - The rpmsg endpoint
+ *   data - The return message
+ *   len  - The return message length
+ *   src  - unknow
+ *   priv - unknow
+ *
+ * Returned Values:
+ *   OK on success; A negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+static int rpmsghci_send_handler(FAR struct rpmsg_endpoint *ept,
+                                  FAR void *data, size_t len,
+                                  uint32_t src, FAR void *priv)
+{
+  FAR struct rpmsghci_server_s *server = NULL;
+  FAR struct bt_driver_s       *btdev  = NULL;
+  FAR struct rpmsghci_data_s   *msg    = data;
+  int                           ret    = OK;
+
+  DEBUGASSERT(priv);
+  server = priv;
+  DEBUGASSERT(server->btdev);
+  btdev = server->btdev;
+
+  wlinfo("rpmsghci_send_handler %d\n", msg->type);
+
+  ret = btdev->send(btdev, msg->type, msg->data,
+                    len - sizeof(struct rpmsghci_data_s) + 1);
+
+  msg->header.result = ret;
+
+  return rpmsg_send(ept, msg, sizeof(struct rpmsghci_header_s));
+}
+
+/****************************************************************************
+ * Name: rpmsghci_ioctl_handler
+ *
+ * Description:
+ *   Rpmsg-HCI ioctl handler.
+ *
+ * Parameters:
+ *   ept  - The rpmsg endpoint
+ *   data - The return message
+ *   len  - The return message length
+ *   src  - unknow
+ *   priv - unknow
+ *
+ * Returned Values:
+ *   OK on success; A negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+static int rpmsghci_ioctl_handler(FAR struct rpmsg_endpoint *ept,
+                                  FAR void *data, size_t len,
+                                  uint32_t src, FAR void *priv)
+{
+  FAR struct rpmsghci_server_s *server = NULL;
+  FAR struct bt_driver_s       *btdev  = NULL;
+  FAR struct rpmsghci_ioctl_s  *msg    = data;
+  int                           ret    = OK;
+
+  DEBUGASSERT(priv);
+  server = priv;
+  DEBUGASSERT(server->btdev);
+  btdev = server->btdev;
+
+  ret = btdev->ioctl(btdev, msg->request,
+                     msg->arglen > 0 ? (unsigned long)msg->buf :
+                     msg->arg);
+
+  msg->header.result = ret;
+
+  return rpmsg_send(ept, msg, sizeof(struct rpmsghci_header_s));
```

Comment:
```suggestion
  return rpmsg_send(ept, msg, len);
```
to avoid the lose of returned value